spark3/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -63,7 +63,8 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(SparkTable.class);
 
-  private static final Set<String> RESERVED_PROPERTIES = Sets.newHashSet("provider", "format", "current-snapshot-id");
+  private static final Set<String> RESERVED_PROPERTIES =
+      Sets.newHashSet("provider", "format", "current-snapshot-id", "location");
```

Comment:
nit: not directly related to this PR, but shall we use `ImmutableSet` for consistency?

Reply:
Fair enough. Made the change.